api/annotate/traces.go

```diff
@@ -0,0 +1,160 @@
+		return
+	}
+
+	var responses []TracesResourceResponse
+	for _, resource := range resources {
+		// choose the annotation key and value according to the telemetry type and action
+		var annotationKey = "logz.io/traces_instrument"
+		value := "true"
+		if resource.Action == "delete" {
+			value = "rollback"
+		}
+
+		annotations := map[string]string{
+			annotationKey: value,
+		}
+
+		// Create the response
+		response := TracesResourceResponse{
+			Name:               resource.Name,
+			Namespace:          resource.Namespace,
+			Kind:               resource.Kind,
+			UpdatedAnnotations: annotations,
+		}
+
+		switch resource.Kind {
+		case "deployment":
+			deployment, err := clientset.AppsV1().Deployments(resource.Namespace).Get(r.Context(), resource.Name, v1.GetOptions{})
+			if err != nil {
+				logger.Error("Error getting deployment", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			for k, v := range annotations {
+				if deployment.Spec.Template.ObjectMeta.Annotations == nil {
+					deployment.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+				}
+				deployment.Spec.Template.ObjectMeta.Annotations[k] = v
+			}
+
+			_, err = clientset.AppsV1().Deployments(resource.Namespace).Update(r.Context(), deployment, v1.UpdateOptions{})
+			if err != nil {
+				logger.Error("Error updating deployment", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			responses = append(responses, response)
+
+		case "statefulset":
+			statefulSet, err := clientset.AppsV1().StatefulSets(resource.Namespace).Get(r.Context(), resource.Name, v1.GetOptions{})
+			if err != nil {
+				logger.Error("Error getting statefulset", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			for k, v := range annotations {
+				if statefulSet.Spec.Template.ObjectMeta.Annotations == nil {
+					statefulSet.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+				}
+				statefulSet.Spec.Template.ObjectMeta.Annotations[k] = v
+			}
+
+			_, err = clientset.AppsV1().StatefulSets(resource.Namespace).Update(r.Context(), statefulSet, v1.UpdateOptions{})
+			if err != nil {
+				logger.Error("Error updating statefulset", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			responses = append(responses, response)
+		}
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(responses)
+}
+
+func validateTracesResourceRequests(resources []TracesResourceRequest) bool {
+	for _, resource := range resources {
+		if !isValidTracesResourceRequest(resource) {
+			return false
+		}
+	}
+	return true
+}
+
+func isValidTracesResourceRequest(req TracesResourceRequest) bool {
+	isValidKind := req.Kind == "statefulset" || req.Kind == "deployment"
+	isValidAction := req.Action == "add" || req.Action == "delete"
```

Comment:
I'd prefer if:
1. we'll move these to be consts (maybe even under common.go)
2. In the function create an array of supported kinds and supported actions.
3. Iterate through those arrays to check validity
4. Use those consts throughout the code